internal/kittehs/strings.go

```diff
@@ -52,3 +53,29 @@ func PadLeft(s string, r rune, n int) string {
 
 	return strings.Repeat(string(r), n-len(s)) + s
 }
+
+// NormalizeURL ensures that the given URL has the right scheme
+// prefix, and no suffix (e.g. path) after the host address.
+func NormalizeURL(rawURL string, secure bool) (string, error) {
+	// Normalize the URL's scheme prefix.
+	scheme := "http://"
```

Comment:
instead of parsing it yourself prior to Parse below, just let the lib parse it, then set the Scheme to whatever you want. Then convert to string.

Reply:
This won't work, if the scheme is missing in the rawURL, url.Parse will think that the host address is actually just a relative path.

Reply:
sg